src/gui/shell.cpp

```diff
@@ -1288,6 +1299,20 @@ void Shell::mousePressEvent(QMouseEvent *ev)
 	mouseClickIncrement(ev->button());
 	neovimMouseEvent(ev);
 }
+
+/** Update Current Locale */
+void Shell::updateCurrentLocale() {
+	m_currentInputMethod = QGuiApplication::inputMethod();
+	if(m_currentInputMethod != nullptr) {
+		m_currentLocale =  m_currentInputMethod->locale();
+	} 
+}
+/** Keyboard Locale has changed */
+void Shell::localeChanged()
+{
```

Comment:
I could be wrong... If this doesn't work, ignore my comment :)

No need for `updateCurrentLocale` or `m_currentInputMethod`.

I think you can write:
```C++
void Shell::localeChanged() noexcept
{
  const QInputMethod* inputMethod{ QGuiApplication::inputMethod() };
  if (!inputMethod)
  {
    qDebug() << "Unexpected: QGuiApplication::inputMethod() returned nullptr!"
    return;
  }

  m_currentLocale = inputMethod->locale();
}
```

Less code, less state.

Reply:
initially i neede to call the logic from that method on 2 places
  - the "change" signal.
  - on startup

the startup part moved into the initializer - so yes, i am going to move it

Reply:
ok that doesnt work out, one needs the ref to the inputMethod for the signal and so on, so i  think  we'll keep it.

Reply:
Ahhh I see the issue, Signal/Slot connection. It should be safe to call QGuiApplicatoin::inputMethod() directly instead.

From my quick read, the function returns a single instance every time:
https://code.woboq.org/qt5/qtbase/src/gui/kernel/qguiapplication.cpp.html#_ZN15QGuiApplication11inputMethodEv

The perf on that function should be good, no need for `m_currentInputMethod`.

If the object returned were to change, we'd have a bug: `connect(m_currentInputMethod, ...` is not updated to the new object.

Patch attached for your reference:
[RemoveCurrentInputMethodMemberPatch.txt](https://github.com/equalsraf/neovim-qt/files/6332135/RemoveCurrentInputMethodMemberPatch.txt)

`git apply RemoveCurrentInputMethodMemberPatch.txt`

Reply:
patch manually applied!